src/beamlime/logging/resources.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+# See comments in ``logging.Logger._src_file``
+# for explanation why we didn't use __file__
+import logging
+import time
+from contextlib import contextmanager
+from pathlib import Path
+from threading import RLock
+from typing import Iterator, NewType
+
+from ..empty_providers import log_providers
+
+TimeStamp = NewType("TimeStamp", str)
+
+
+@log_providers.provider
+def create_time_stamp() -> TimeStamp:
+    """
+    Creates a timestamp with the format, ``{cur_timestamp}--{h_time}``.
+    ``htime`` is as an extra information only for the human users,
+    and the ``cur_timestamp`` should be used by the applications or modules.
+    """
+    from datetime import datetime
+
+    cur_timestamp = round(time.time())
+    h_time = datetime.fromtimestamp(cur_timestamp).strftime("%Y-%m-%d-%H-%M-%S")
```

Comment:
Can you change this to use the ISO format?

```suggestion
    h_time = datetime.fromtimestamp(cur_timestamp).isoformat()
```

And ideally even with timezone information. I'm weary of times without timezone information as they can easily be misinterpreted.
You could use the local timezone if you are willing to use a package like dateutil. But I think it is enough to show everything in UTC using:

```suggestion
    from datetime import datetime, timezone

    now = datetime.now(tz=timezone.utc)
    cur_timestamp = round(now.timestamp())
    h_time = now.isoformat()
```

But in general, why do we need the timestamp? If we use the full ISO format, it can be parsed back into a `datetime`, esp. if we force it to be in UTC.